src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -160,6 +185,59 @@ class TextualOccurenceAtPositionRequest extends OccurenceAtPositionRequest {
 		return super.isValid(model, selection, decorations);
 	}
 }
+class MultiModelOccurenceRequest extends OccurenceAtPositionRequest {
+	private readonly _providers: LanguageFeatureRegistry<MultiDocumentHighlightProvider>;
+	private readonly _otherModels: ITextModel[];
+
+	constructor(model: ITextModel, selection: Selection, wordSeparators: string, providers: LanguageFeatureRegistry<MultiDocumentHighlightProvider>, otherModels: ITextModel[]) {
+		super(model, selection, wordSeparators);
+		this._providers = providers;
+		this._otherModels = otherModels;
+	}
+
+	protected override _compute(model: ITextModel, selection: Selection, wordSeparators: string, token: CancellationToken): Promise<Map<URI, DocumentHighlight[]>> {
+		return getOccurrencesAcrossMultipleModels(this._providers, model, selection.getPosition(), wordSeparators, token, this._otherModels).then(value => {
+			if (!value) {
+				return new Map<URI, DocumentHighlight[]>();
+			}
+			return value;
+		});
+	}
+}
+
+class TextualMultiModelOccurenceRequest extends OccurenceAtPositionRequest {
```

Comment:
`TextualMultiModelOccurenceRequest` and `TextualOccurenceAtPositionRequest` are almost identical, I wonder if `TextualOccurenceAtPositionRequest` is  `TextualMultiModelOccurenceRequest` with `otherModels: []`.